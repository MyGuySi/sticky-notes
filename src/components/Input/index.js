// @noflow
// TODO: - Fix issue with Flow type for 'otherProps'

import React from 'react';
import styles from './styles.module.css';

type Props = { label: string }

const Input = React.forwardRef((props: Props, ref) => {
  const { label,...otherProps } = props

  return (
    <div className={styles.wrapper}>
      {label && (
        <label className={styles.label}>{label}:</label>
      )}
      <input ref={ref} className={styles.input} {...otherProps} />
    </div>
  )
})

export default Input;