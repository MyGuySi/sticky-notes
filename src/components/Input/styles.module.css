.wrapper {
  display: flex;
  flex-direction: column;
}

.label {}

.input {
  font-size: 1rem;
  border-radius: 0.25rem;
  border: 0;
  padding: 0.5rem 0.75rem;
  border: 1px solid #dfdfdf;
  box-shadow: inset 0 0 4px rgba(0, 0, 0, 0.1);
}