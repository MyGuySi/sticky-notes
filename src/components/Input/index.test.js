// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import Input from './index'

test('Renders correctly', () => {
  const component = renderer.create(
    <Input label='Name' type='text' placeholder='e.g. John Smith' onChange={e => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});