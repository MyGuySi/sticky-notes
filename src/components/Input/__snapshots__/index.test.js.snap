// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="wrapper"
>
  <label
    className="label"
  >
    Name
    :
  </label>
  <input
    className="input"
    onChange={[Function]}
    placeholder="e.g. John Smith"
    type="text"
  />
</div>
`;
